src/quest.cpp

```diff
@@ -1138,22 +1182,33 @@ SPECIAL(johnson)
         return TRUE;
       }
       
+      new_q = new_quest(johnson, ch);
```

Comment:
Is there a way we can cache what quest the person is trying to get on? We're doing a lot of calls to `new_quest`, which means a full iteration of the quest table and a list sort for each one. Sticking it on a temporary var for the character doing the talking seems preferable.

Reply:
I thought about this myself. I cached it as variable and there are 3 calls to `new_quest()` in the code and 2 of them are mutually exclusive (yes or no).  So basically we only do 2 calls, one on start and one on accept or decline. Considering these do not happen very often (just when getting a run) I thought it would be acceptable. Vector containers are very very fast to sort (lightning fast at that data size) and memory efficient as well. But we could cache it on the character, to save 1 call if you think it's too heavy.

Reply:
Good point, we can skip caching the result then.

Reply:
I thought of a way to actually save us the second call without caching it on the character even though we initialize `new_q` in a switch case. https://github.com/luciensadi/AwakeMUD/pull/582/commits/e280b62b9f227ea1942fe8add9ad80057e03ed2d